src/main/java/de/tum/in/www1/artemis/repository/CourseRepository.java

```diff
@@ -58,15 +58,17 @@ public interface CourseRepository extends JpaRepository<Course, Long> {
             """)
     List<Course> findAllActive(@Param("now") ZonedDateTime now);
 
-    @EntityGraph(type = LOAD, attributePaths = { "lectures", "lectures.attachments", "exams" })
+    @EntityGraph(type = LOAD, attributePaths = { "lectures", "lectures.attachments" })
     @Query("""
             SELECT DISTINCT c FROM Course c
-            WHERE (c.startDate <= :now
-            	OR c.startDate IS NULL)
-            AND (c.endDate >= :now
-            	OR c.endDate IS NULL)
+            LEFT JOIN FETCH c.exams exams
+            LEFT JOIN FETCH exams.registeredUsers registeredUsers
```

Comment:
```suggestion
            LEFT JOIN FETCH c.exams exams
            LEFT JOIN exams.registeredUsers registeredUsers
```
I don’t think the users are needed in the code somewhere. I think a join without fetch should work?